sdk/core/core/inc/az_span.h

```diff
@@ -35,44 +38,45 @@ typedef struct
 /********************************  SPAN GETTERS */
 
 /**
- * @brief convenient way to get the pointer from one span
+ * @brief az_span_ptr returns the span byte buffer's starting memory address
  *
  */
 AZ_NODISCARD AZ_INLINE uint8_t* az_span_ptr(az_span span) { return span._internal.ptr; }
 
 /**
- * @brief convenient way to get the length from one span
+ * @brief az_span_length returns the span byte buffer's length (number of bytes in use from the
+ * buffer's start)
  *
  */
 AZ_NODISCARD AZ_INLINE int32_t az_span_length(az_span span) { return span._internal.length; }
 
 /**
- * @brief convenient way to get the capacity from one span
+ * @brief az_span_capacity returns the span byte buffer's capacity (in bytes)
```

Comment:
I struggled with this too. The span is not the byte buffer; a span refers to a subset of a byte buffer. I don't have an easy way to say this that just flows off the tongue. But, I'm open to suggestions. I don't think it is right to say "Do not access the memory beyond this capacity." because, when using a span and its functions, this is impossible. And also, a span could be to a subset of a byte buffer and the customer might know that accessing bytes past the span's capacity IS OK. 

Saying that length is always <= capacity is fine with me.

Reply:
> And also, a span could be to a subset of a byte buffer and the customer might know that accessing bytes past the span's capacity IS OK.

Hmm... interesting. I understand that the caller can do this, but should they be? Isn't it considered potentially bad practice to go outside the bounds of the span? How do you reason about what the scope of the "original" span is to know it is safe to go outside of its capacity bounds?

Reply:
It is common to create span over the large byte buffer. And then to create span slices with the buffer. So, the customer is working with multiple span instances simultaneously. The "proper" practice would be to throw away the slice span and go back to the original and then not access bytes beyond the out span's capacity. 

So, yes, you're right that they shouldn't do this but I don't think we should explicitly say so. In reality, the only way they could do this is by getting the span's pointer and then doing whatever they want to with it. Really, they should not get the pointer as that is effectively unsafe code. If they just stick with ALL our other function, then the manipulations are safe. I'd rather discourage calling az_span_ptr.